### PR TITLE
Fix all vulnerabilities by updating packages

### PR DESCRIPTION
**Pull request by sfc-gh-rcheng**
Minimum JDBC version: 3.13.31, however needs to upgrade: 
- com.nimbusds:numbus-jose-jwt:9.2 (uses vulnerable net.minidev:json-smart:2.4.7) 
- Vulnerable com.google.guava:guava:30.0-jre -> 32.0.1-jre

Need to exclude org.apache.hadoop.thirdparty:hadoop-shaded-guava:1.1.1, but fails FlushServiceTest

### Changes
**❌ net.minidev:json-smart**
- Requires version update: 2.4.7 -> 2.4.9
- Search file: jsonaware
- Changes
	- net.snowflake:snowflake-jdbc:  3.13.30 -> 3.13.31
	 	- ✅ net.minidev:json-smart: 2.4.8 -> [2.4.9](https://github.com/snowflakedb/snowflake-jdbc/commit/aaf787e72adf73333caf8ce6aa2baae9fa1b0a1e)
	 	- ❌ NEEDS UPGRADE [v3.13.33](https://github.com/snowflakedb/snowflake-jdbc/blob/v3.13.33/pom.xml#L286) com.nimbusds:numbus-jose-jwt:9.21 (net.minidev:json-smart:2.4.7) 

**✅ com.google.protobuf:protobuf-java**
- Requires version update: 3.7.1 -> 3.16.3
- Search file: blockingrpcchannel
- Changes
	- 🆗 (no-op) com.google.protobuf:protobuf-java:3.19.6
	- ✅ exclude org.apache.hadoop.thirdparty:hadoop-shaded-protobuf_3_7

**❌ com.google.guava:guava**
- Requires version update: 30.1.1-jre -> 32.0.1
	- "Even though the security vulnerability is fixed in version 32.0.0, we recommend using version 32.0.1 as version 32.0.0 breaks some functionality under Windows." [CVE](https://nvd.nist.gov/vuln/detail/CVE-2023-2976)
- Search file: extraobjectmethodsforweb
- Changes
	- 🆗 (no-op) com.google.guava:guava:32.0.1-jre
	- ❌ exclude (but fails unit tests) org.apache.hadoop.thirdparty:hadoop-shaded-guava:1.1.1 (no upgrade)
		- com.google.guava:guava:[31.1-jre](https://github.com/apache/hadoop-thirdparty/blob/trunk/pom.xml#L98)
	- ❌ NEEDS UPGRADE net.snowflake:snowflake-jdbc: 3.13.30 -> 3.13.33
		- com.google.guava:guava:30.0-jre -> [31.1-jre](https://github.com/snowflakedb/snowflake-jdbc/commit/81eb493c10b0ae0ffbcea35828d0dda324dd044b)
		
		
**✅ com.fasterxml.jackson.core:jackson-databind**
- Requires version update: 2.13.2.2 -> 2.13.4
- Search file: propertynamingstrategies
- Changes
 	- 🆗 (no-op) com.fasterxml.jackson.core:jackson-databind:2.14.0
	- 🆗 (no-op) net.snowflake:snowflake-jdbc:3.13.30 on [2.13.4.2](https://github.com/snowflakedb/snowflake-jdbc/blob/v3.13.30/pom.xml#L45)
	- ✅ org.apache.parquet:parquet-jackson:1.12.3 -> 1.13.1 (in previous [pr](https://github.com/snowflakedb/snowflake-ingest-java/commit/2af18ec348c892f8d61899c4a5c2817b39d4ceed))
		- com.fasterxml.jackson.core:jackson-databind: 2.13.2.2 -> [2.13.4.2](https://github.com/apache/parquet-mr/blob/master/pom.xml#L70)

### CODE REVIEW

**Review by sfc-gh-tjones — APPROVED** [review]
Nice! Were these vulnerabilities identified via Blackduck or Synk?

**Comment by sfc-gh-rcheng** [comment]
> Nice! Were these vulnerabilities identified via Blackduck or Synk?

identified with Wiz, which is the closest we have to Blackduck

File: pom.xml [hunk 1]
```diff
@@ -64,7 +64,7 @@
     <shadeBase>net.snowflake.ingest.internal</shadeBase>
     <slf4j.version>1.7.36</slf4j.version>
     <snappy.version>1.1.10.1</snappy.version>
-    <snowjdbc.version>3.13.30</snowjdbc.version>
+    <snowjdbc.version>3.13.33</snowjdbc.version>
```
Comment (sfc-gh-tzhang) [inline pom.xml:67]: I don't see all the changes as described in the description, did i miss anything?

Comment (sfc-gh-xhuang) [reply pom.xml:67]: I thought we weren't going to upgrade JDBC. Is upgrading JDBC required for fixing these vulns?

Comment (sfc-gh-rcheng) [reply pom.xml]: We need 3.13.31 at a minimum to resolve the json-smart vulnerability, however there are a couple of vulnerable packages I've found in the latest 3.13.33 release

**Comment by sfc-gh-rcheng** [comment]
wiz scan of previous pr came back clean, these upgrades are not necessary

**Pull request closed without merging**